### PR TITLE
perf: Remove unneeded GetKey calls to the LRU cache

### PR DESCRIPTION
**Pull request by ValarDragon**
Remove some extra `GetKey` calls within the LRU cache code. GetKey calls took 1.5s on an Osmosis 2000 block long block sync. This PR should shave off .45s of that.

Though ideally we get the LRU code rethought / adapted more, to make it generally lower overhead. (Its currently 13.5 s on a 2000 block sync!)

### CODE REVIEW

**Review by kocubinski — APPROVED** [review]

**Review by cool-develope — APPROVED** [review]

**Comment by tac0turtle** [comment]
@mergifyio backport release/v1.x.x

**Comment by odeke-em** [comment]
@ValarDragon this isn't fully the correct optimization despite eliminating the extra call to node.Key() while then also keep the original intended optimization or zero byteslice->string conversion in the case of already existing keys when looking im the map. In short it now it incurs the byteslice->string conversion in all cases, but could have been faster if it had been:

```go
key := node.GetKey()
if value, ok := c.dict[string(key)]; ok {
   ...
}

c.dict[string(key)] = value
```

File: cache/cache.go [hunk 1]
```diff
@@ -96,8 +97,9 @@ func (c *lruCache) Len() int {
 }
 
 func (c *lruCache) Remove(key []byte) Node {
-	if elem, exists := c.dict[string(key)]; exists {
-		return c.remove(elem)
+	keyS := string(key)
+	if elem, exists := c.dict[keyS]; exists {
```
Comment (odeke-em) [inline cache/cache.go:101]: This undoes the zero byteslice->string Go map lookup optimization.

Comment (ValarDragon) [reply cache/cache.go:101]: Oh wow, didn't know about this compiler optimization.

**Pull request merged**